android/src/main/java/com/expensify/livemarkdown/MarkdownUtils.java

```diff
@@ -66,6 +68,62 @@ public void setMarkdownStyle(@NonNull MarkdownStyle markdownStyle) {
     mMarkdownStyle = markdownStyle;
   }
 
+  private List<MarkdownRange> splitRangesOnEmojis(List<MarkdownRange> markdownRanges) {
+    List<MarkdownRange> emojiRanges = new ArrayList<>();
+    for (MarkdownRange range : markdownRanges) {
+      if (range.type.equals("emoji")) {
+        emojiRanges.add(range);
+      }
+    }
+
+    for (int i = 0; i < markdownRanges.size(); i++) {
+      MarkdownRange range = markdownRanges.get(i);
+      if (!range.type.equals("italic") && !range.type.equals("strikethrough")) {
+        continue;
+      }
+
+      for (int j = 0; j <  emojiRanges.size(); j++) {
+        MarkdownRange emojiRange = emojiRanges.get(j);
+        if (emojiRange.start >= range.start && emojiRange.end <= range.end) {
+          // Split range
+          MarkdownRange startRange = new MarkdownRange(range.type, range.start, emojiRange.start, range.depth);
+          MarkdownRange endRange =  new MarkdownRange(range.type, emojiRange.end, range.end, range.depth);
+          markdownRanges.add(i + 1, endRange);
+          markdownRanges.add(i + 1, startRange);
+          markdownRanges.remove(i);
+          i += 1;
+          range = endRange;
+        } else {
+          break;
+        }
+      }
+    }
+    return markdownRanges;
+  }
+
+
+  private MarkdownRange[] parseRanges(String rangesJSON) {
+    try {
+      List<MarkdownRange> markdownRanges = new ArrayList<>();
+      JSONArray ranges = new JSONArray(rangesJSON);
+      for (int i = 0; i < ranges.length(); i++) {
+        JSONObject range = ranges.getJSONObject(i);
+        String type = range.getString("type");
+        int start = range.getInt("start");
+        int length = range.getInt("length");
+        int depth = range.optInt("depth", 1);
+        int end = start + length;
+        markdownRanges.add(new MarkdownRange(type, start, end, depth));
+      }
+      List<MarkdownRange> splittedRanges = splitRangesOnEmojis(markdownRanges);
+      return splittedRanges.toArray(new MarkdownRange[0]);
```

Comment:
Why can't we just return `List<MarkdownRange>`?

Reply:
We can, changed